MvvmCross/Core/Core/ViewModels/NotifyTask.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Honours to Stephen Cleary for this file's code
+
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+
+namespace MvvmCross.Core.ViewModels
+{
+    /// <summary>
+    /// Watches a task and raises property-changed notifications when the task completes.
+    /// </summary>
+    public sealed class NotifyTask : INotifyPropertyChanged
```

Comment:
Maybe we should name this `MvxNotifyTask`? That would make it a bit more logical to also use outside the new viewmodel loading.